src/main/java/tuteez/logic/commands/EditCommand.java

```diff
@@ -97,18 +100,34 @@ public CommandResult execute(Model model) throws CommandException {
 
         Optional<Set<Lesson>> lessons = editPersonDescriptor.getLessons();
         if (lessons.isPresent()) {
-            for (Lesson newLesson : lessons.get()) {
-                if (personToEdit.isLessonScheduled(newLesson)) {
-                    continue;
-                }
-
-                if (model.isClashingWithExistingLesson(newLesson)) {
-                    String logMessage = String.format("Student: %s | Lessons: %s "
-                            + "| Conflict: Clashes with another student's lesson",
-                            editedPerson.getName(), editedPerson.getLessons().toString());
-                    logger.info(logMessage);
-                    throw new CommandException(MESSAGE_DUPLICATE_LESSON);
-                }
+            Map<Person, ArrayList<Lesson>> resultMap = new HashMap<>();
+            for (Lesson newlesson: lessons.get()) {
+                assert newlesson != null;
+                Map<Person, ArrayList<Lesson>> clashingLessons = model.getClashingLessons(newlesson);
+
+                clashingLessons.forEach((person, lessonArr) -> {
+                    /* Operation is safe because we do not allow student's with the exact same name */
+                    if (!editedPerson.hasSameName(person)) {
+                        System.out.println(editedPerson + " is edited");
+                        System.out.println(person + " is not edited");
+                        resultMap.computeIfAbsent(person, k -> new ArrayList<>()).addAll(lessonArr);
+                    }
+                });
+            }
+
+            if (!resultMap.isEmpty()) {
+                String logMessage = String.format("Student: %s | Lessons: %s "
+                        + "| Conflict: Clashes with another student's lesson",
+                        editedPerson.getName(), editedPerson.getLessons().toString());
+                logger.info(logMessage);
+                StringBuilder sb = new StringBuilder(MESSAGE_CLASHING_LESSON).append("\n");
```

Comment:
Perhaps use a more direct name? sb can be confusing

Reply:
Okay ill change it!!